src/MAUI/Maui.Samples/Samples/Layers/PlayKmlTours/PlayKmlTours.xaml.cs

```diff
@@ -129,33 +129,52 @@ private void Tour_PropertyChanged(object sender, PropertyChangedEventArgs e)
             switch (_tourController.Tour.TourStatus)
             {
                 case KmlTourStatus.Completed:
-                case KmlTourStatus.Initialized:
-                    PlayButton.IsEnabled = true;
-                    PauseButton.IsEnabled = false;
+                    PlayPauseButton.IsEnabled = true;
+                    ResetButton.IsEnabled = false;
+                    PlayPauseButton.Text = "Play";
+
+                    // Return to the initial viewpoint to visually indicate the tour being over.
+                    MySceneView.SetViewpointAsync(MySceneView.Scene.InitialViewpoint);
                     break;
 
-                case KmlTourStatus.Paused:
-                    PlayButton.IsEnabled = true;
-                    PauseButton.IsEnabled = false;
-                    ResetButton.IsEnabled = true;
+                case KmlTourStatus.Initialized:
+                    PlayPauseButton.IsEnabled = true;
+                    ResetButton.IsEnabled = false;
+                    PlayPauseButton.Text = "Play";
                     break;
 
                 case KmlTourStatus.Playing:
                     ResetButton.IsEnabled = true;
-                    PlayButton.IsEnabled = false;
-                    PauseButton.IsEnabled = true;
+                    PlayPauseButton.IsEnabled = true;
+                    PlayPauseButton.Text = "Pause";
+                    break;
+
+                case KmlTourStatus.Paused:
+                    PlayPauseButton.Text = "Play";
                     break;
             }
         }
 
-        // Play the tour when the button is pressed.
-        private void Play_Clicked(object sender, EventArgs e) => _tourController?.Play();
-
-        // Pause the tour when the button is pressed.
-        private void Pause_Clicked(object sender, EventArgs e) => _tourController?.Pause();
+        // Play and pause the tour when the button is pressed.
+        private void PlayPause_Clicked(object sender, EventArgs e)
+        {
+            if (PlayPauseButton.Text == "Play")
```

Comment:
We shouldn't couple our sample functionality to label text like this. 

A better solution would be to use a private boolean variable `_isPlaying` that is updated on play/pause and checked in situations like this.